Tasks/SonarQubePreBuild/SonarQubePreBuild.ps1

```diff
@@ -27,30 +27,17 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 
 . $PSScriptRoot/Common/SonarQubeHelpers/SonarQubeHelper.ps1
 
-if ( (IsPrBuild) -and ((GetTaskContextVariable "DisableSQAnalysisOnPrBuilds") -eq "true"))
+# During PR builds only an "issues mode" analysis is allowed. The resulting issues are posted as code review comments. 
+# The feature can be toggled by the user and is OFF by default.  
+if (ShouldExitOnPRBuild)
 {
-	Write-Host "DisableSQAnalysisOnPrBuilds is set and this is a PR build - ignoring the analysis tasks"	
-	return
+    Write-Host "SonarQube analysis is disabled during builds triggered by pull requests. Set a build variable named 'SQPullRequestBot' to 'true' to have the task post code analysis issues as comments in the PR. More information at http://go.microsoft.com/fwlink/?LinkID=786316"
```

Comment:
Is it necessary to write the same message in both the _begin_ and _end_ tasks - isn't just the _begin_ task enough?

If you think it is necessary to issue the message twice, could you avoid the code duplication by pushing all of the logic into the helper method and have the helper method write the message then exit?
Also, is all of the duplication necessary? If you're using _exit_ couldn't you put all